gstreamer/src/query.rs

```diff
@@ -270,10 +270,154 @@ pub enum QueryView {
+            QueryView::Seeking(ref seeking) => seeking.into(),
+            QueryView::Segment(ref segment) => segment.into(),
+            QueryView::Convert(ref convert) => convert.into(),
+            QueryView::Formats(ref formats) => formats.into(),
+            QueryView::Buffering(ref buffering) => buffering.into(),
+            QueryView::Custom(ref custom) => custom.into(),
+            QueryView::Uri(ref uri) => uri.into(),
+            QueryView::Allocation(ref allocation) => allocation.into(),
+            QueryView::Scheduling(ref scheduling) => scheduling.into(),
+            QueryView::AcceptCaps(ref accept_caps) => accept_caps.into(),
+            QueryView::Caps(ref caps) => caps.into(),
+            QueryView::Drain(ref drain) => drain.into(),
+            QueryView::Context(ref context) => context.into(),
+            QueryView::Other(ref other) => other.into(),
+            _ => unimplemented!("Converting QueryView into Query for unknown query type"),
+        }
+    }
+}
+
+impl<'a> Into<&'a mut Query> for &'a mut QueryView {
+    fn into(self) -> &'a mut Query {
+        match *self {
+            QueryView::Position(ref mut position) => position.into(),
+            QueryView::Duration(ref mut duration) => duration.into(),
+            QueryView::Latency(ref mut latency) => latency.into(),
+            QueryView::Jitter(ref mut jitter) => jitter.into(),
+            QueryView::Rate(ref mut rate) => rate.into(),
+            QueryView::Seeking(ref mut seeking) => seeking.into(),
+            QueryView::Segment(ref mut segment) => segment.into(),
+            QueryView::Convert(ref mut convert) => convert.into(),
+            QueryView::Formats(ref mut formats) => formats.into(),
+            QueryView::Buffering(ref mut buffering) => buffering.into(),
+            QueryView::Custom(ref mut custom) => custom.into(),
+            QueryView::Uri(ref mut uri) => uri.into(),
+            QueryView::Allocation(ref mut allocation) => allocation.into(),
+            QueryView::Scheduling(ref mut scheduling) => scheduling.into(),
+            QueryView::AcceptCaps(ref mut accept_caps) => accept_caps.into(),
+            QueryView::Caps(ref mut caps) => caps.into(),
+            QueryView::Drain(ref mut drain) => drain.into(),
+            QueryView::Context(ref mut context) => context.into(),
+            QueryView::Other(ref mut other) => other.into(),
+            _ => unimplemented!("Converting QueryView into Query for unknown query type"),
+        }
+    }
+}
+
 macro_rules! declare_concrete_query(
     ($name:ident) => {
         pub struct $name(Query);
 
+        impl $name {
+            pub unsafe fn into_ptr(self) -> *mut ffi::GstQuery {
+                self.0.into_ptr()
+            }
+        }
+
+        impl Into<QueryView> for $name {
+            fn into(self) -> QueryView {
+                QueryView::$name($name(self.0))
+            }
+        }
+
+        impl Into<Query> for $name {
```

Comment:
Usually you would implement `From`, not `Into`. It's slightly more general AFAIU, and you can implement it here